### PR TITLE
Fix sendStream() failing if it can't complete immediately.

### PR DESCRIPTION
**Pull request by kentonv**

### CODE REVIEW

File: c++/src/kj/async-io-test.c++ [hunk 1]
```diff
@@ -238,6 +238,49 @@ TEST(AsyncIo, CapabilityPipe) {
...
+      ++nonBlockedCount;
+      promise.wait(io.waitScope);
+    } else {
+      // Send blocked! Let's continue with this promise then!
+      endpoint1 = kj::mv(pipe2.ends[1]);
+      break;
+    }
+  }
+
+  KJ_DBG(nonBlockedCount);
```
Comment (harrishancock) [inline c++/src/kj/async-io-test.c++]: Vestigial KJ_DBG

Comment (kentonv) [reply c++/src/kj/async-io-test.c++]: oops, fixed. (It printed ~300 FWIW.)

Comment (kentonv) [reply c++/src/kj/async-io-test.c++]: Or 278, to be precise. A weirdly non-round number, and smaller than I would have expected.

Comment (harrishancock) [reply c++/src/kj/async-io-test.c++]: Nice, I was curious. I was expecting it to be in the thousands.

**Comment by kentonv** [comment]
@harrishancock Yeah I think that test also failed before I updated the PR, and I re-ran it, and then it passed. Seems it's a bit flaky. :/

File: c++/src/kj/async-io-unix.c++ [hunk 2]
```diff
@@ -216,7 +216,7 @@ public:
       return downcast<AsyncStreamFd>(*stream).fd;
     };
     auto promise = writeInternal(data, moreData, fds);
-    return promise.attach(kj::mv(fds));
+    return promise.attach(kj::mv(fds), kj::mv(streams));
```
Comment (harrishancock) [inline c++/src/kj/async-io-unix.c++]: BTW, I must have read this line of code 5 times in the last 2 days and completely missed this.

**Comment by jclee** [comment]
> Looks transient, though? FYI @jclee

That's unfortunate...  That test is based on the one above it, which was formerly disabled for both Windows and Mac due to flakiness, until I found that adding an extra poll() seemed to make the test pass reliably.  I was hoping the same would be true for this test, but I guess it isn't.  Since two connections are involved, I guess I could try adding a second poll() to see if it makes the test reliable, under the assumption that N connections opening may require up to N trips around the event loop.  Or we could just disable the test entirely on Windows, until we work out what is actually going wrong?

File: c++/src/kj/async-io-unix.c++ (hunk 2 above)
Comment (kentonv) [reply c++/src/kj/async-io-unix.c++]: I missed it a few times too. Didn't dawn on me until I read the calling code and noticed that ownership of the streams array was being passed in.

**Pull request merged**